packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -68,13 +84,32 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         }
     }
 
+    private async _handleClick(
+        buyNowInitializeOptions: PaypalCommerceButtonInitializeOptions['buyNowInitializeOptions'],
+    ): Promise<void> {
+        if (buyNowInitializeOptions && typeof buyNowInitializeOptions.getBuyNowCartRequestBody === 'function') {
+            const cartRequestBody = buyNowInitializeOptions.getBuyNowCartRequestBody();
+
+            if (!cartRequestBody) {
+                throw new MissingDataError(MissingDataErrorType.MissingCart);
+            }
+
+            try {
+                const response = await this._cartRequestSender.createBuyNowCart(cartRequestBody);
+
+                this._buyNowCartId = response.body.id;
+            } catch (error) {
+                throw new BuyNowCartCreationError();
+            }
+        }
+    }
+
     private async _createOrder(initializesOnCheckoutPage?: boolean): Promise<string> {
-        const state = this._store.getState();
-        const cart = state.cart.getCartOrThrow();
+        const cartId = this._buyNowCartId || this._store.getState().cart.getCartOrThrow().id;
```

Comment:
Can we check here if this is a buyNowCart order or a regular cart order?

Reply:
I don't think we should, because _buyNowCartId should be undefined for regular cart order